### PR TITLE
Photos Picker: Create data layer for creating/fetching session and media items

### PR DESCRIPTION
**Pull request by bogiii**
Related to #

## Proposed Changes

* Updated Redux store to support Google Photos Picker Session
* Introduced `session` query and mutation hooks
* Introduced `media-items` query
* Created high-order component providing session and method for session creation

## Why are these changes being made?

*

## Testing Instructions

* This PR doesn't introduce data layer integration, it will in the next PR
* It's OK to do code review only
* or to wrap the `client/my-sites/media-library/content.jsx` with `withGooglePhotosPickerSession` like:
  * `withGooglePhotosPickerSession( localize( MediaLibraryContent ) )` and manually run the mutation/query hooks

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/data/media/use-google-photos-picker-media-items-query.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { useQuery } from '@tanstack/react-query';
+import wp from 'calypso/lib/wp';
+
+interface Thumbnails {
+	large: string;
+	medium: string;
+	thumbnail: string;
+}
```
Comment (ouikhuan) [inline client/data/media/use-google-photos-picker-media-items-query.ts]: Missing one `post-thumbnail` here.

File: client/data/media/use-google-photos-picker-media-items-query.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+				fmt_dvd: string;
+				fmt_sd: string;
+		  };
+}
+
+interface ResponseData {
+	found: number;
+	media: Media[];
+	meta?: {
+		next_page: boolean;
```
Comment (ouikhuan) [inline client/data/media/use-google-photos-picker-media-items-query.ts:37]: There's also an `account` under meta, which contains two keys:

```
account:{ 
image:string
name:string
}
```

**Review by ouikhuan — APPROVED** [review]
Other than the type missing, and the new endpoint changes we already mentioned, everything else looks good! Thank you!

File: client/data/media/use-google-photos-picker-media-items-query.ts (hunk 1 above)
Comment (bogiii) [reply client/data/media/use-google-photos-picker-media-items-query.ts]: To double-check with you, `post_thumbnail` or `post-thumbnail`? Dash or underscore in between?

File: client/data/media/use-google-photos-picker-media-items-query.ts (hunk 2 above)
Comment (bogiii) [reply client/data/media/use-google-photos-picker-media-items-query.ts:37]: Added, thanks!

Comment (bogiii) [reply client/data/media/use-google-photos-picker-media-items-query.ts:37]: I just found it's with dash: `post-thumbnail`

File: client/data/media/use-google-photos-picker-media-items-query.ts (hunk 1 above)
Comment (bogiii) [reply client/data/media/use-google-photos-picker-media-items-query.ts]: I just found it's with dash: post-thumbnail

**Pull request merged**